### PR TITLE
I have modified the Bowtie mapper class because the index for bowtie2…

### PR DESCRIPTION
**Pull request by Acribbs**
… was not being set and was picking up the bowtie index dir

### CODE REVIEW

File: CGATPipelines/PipelineMapping.py [hunk 1]
```diff
@@ -3044,12 +3044,21 @@ def mapper(self, infiles, outfile):
                 nfiles -= 2
             else:
                 raise ValueError("unexpected number of files")
-            index_prefix = "%(bowtie_index_dir)s/%(genome)s_cs"
+            if exexutable == "bowtie":
```
Comment (sebastian-luna-valero) [inline CGATPipelines/PipelineMapping.py]: should this be `executable` instead?

File: CGATPipelines/PipelineMapping.py [hunk 2]
```diff
@@ -3044,12 +3044,21 @@ def mapper(self, infiles, outfile):
...
+                index_prefix = "%(bowtie_index_dir)s/%(genome)s_cs"
+            else:
+                index_prefix = "%(bowtie2_index_dir)s/%(genome)s_cs"
         elif self.datatype == "fasta":
             data_options.append("-f")
-            index_prefix = "%(bowtie_index_dir)s/%(genome)s"
+            if exexutable == "bowtie":
```
Comment (sebastian-luna-valero) [inline CGATPipelines/PipelineMapping.py]: should this be `executable` instead?

File: CGATPipelines/PipelineMapping.py [hunk 3]
```diff
@@ -3044,12 +3044,21 @@ def mapper(self, infiles, outfile):
...
+                index_prefix = "%(bowtie_index_dir)s/%(genome)s"
+            else:
+                index_prefix = "%(bowtie2_index_dir)s/%(genome)s"
         else:
-            index_prefix = "%(bowtie_index_dir)s/%(genome)s"
+            if exexutable == "bowtie":
```
Comment (sebastian-luna-valero) [inline CGATPipelines/PipelineMapping.py]: should this be `executable` instead?

**Review by sebastian-luna-valero — CHANGES_REQUESTED** [review]
Thanks @Acribbs 

Please see my comments before merging

File: CGATPipelines/PipelineMapping.py [hunk 4]
```diff
@@ -3044,12 +3044,21 @@ def mapper(self, infiles, outfile):
                 nfiles -= 2
             else:
                 raise ValueError("unexpected number of files")
-            index_prefix = "%(bowtie_index_dir)s/%(genome)s_cs"
+            if exexutable == "bowtie":
+                index_prefix = "%(bowtie_index_dir)s/%(genome)s_cs"
+            else:
```
Comment (sebastian-luna-valero) [inline CGATPipelines/PipelineMapping.py]: I would even do `elif executable = "bowtie2"` and raise an exception in the `else`, as a sanity check

**Review by sebastian-luna-valero — COMMENTED** [review]
Additional idea.

File: CGATPipelines/PipelineMapping.py (hunk 4 above)
Comment (Acribbs) [reply CGATPipelines/PipelineMapping.py]: Yes I agree with you. I will make the changes.

**Comment by Acribbs** [comment]
@sebastian-luna-valero  I have made the changes you suggest. Thanks

**Comment by Acribbs** [comment]
@sebastian-luna-valero Done, many thanks

**Review by sebastian-luna-valero — APPROVED** [review]

**Comment by sebastian-luna-valero** [comment]
Thanks @Acribbs

**Pull request merged**